### PR TITLE
New ES

### PR DESCRIPTION
**Pull request by bentsi**
@roydahan

### CODE REVIEW

File: sdcm/results_analyze.py [hunk 1]
```diff
@@ -303,8 +305,7 @@ def check_regression(self, test_id, is_gce=False):
                        'hits.hits._source.results.stats_average',
                        'hits.hits._source.results.stats_total',
                        'hits.hits._source.versions']
-        tests_filtered = self._es.search(index=self._index, doc_type=test_type, q=query, filter_path=filter_path,
-                                         size=self._limit)
+        tests_filtered = self._es.search(index=self._es_index, q=query, filter_path=filter_path, size=self._limit)
```
Comment (larisau) [inline sdcm/results_analyze.py]: this search will return all the tests in the index, but we should filter certain test types only.
Yuo need to add the test_name to the query

File: sdcm/microbenchmarking.py [hunk 2]
```diff
@@ -20,7 +20,7 @@
 ch.setFormatter(formatter)
 logger.addHandler(ch)
 
-ES_INDEX = 'test13_microbenchmarking'
+ES_INDEX = 'microbenchmarking'
```
Comment (amoskong) [inline sdcm/microbenchmarking.py]: If it's not a fixed name, how about move it to config file?

File: sdcm/results_analyze.yaml [hunk 3]
```diff
@@ -1,5 +1,7 @@
-es_url: 'http://ec2-54-174-18-71.compute-1.amazonaws.com:9200/'
+es_url: 'http://ec2-54-174-18-71.compute-1.amazonaws.com:9200'
 es_index: 'performanceregression'
```
Comment (amoskong) [inline sdcm/results_analyze.yaml]: The parameter `es_index`  isn't used.

**Comment by bentsi** [comment]
Rebased on latest master and fixed all conflicts. 
Please review ASAP.
We need to merge this monster!!!

File: sdcm/results_analyze.py (hunk 1 above)
Comment (roydahan) [reply sdcm/results_analyze.py]: @bentsi ?

File: tests/janus-graph.yaml [hunk 4]
```diff
@@ -28,7 +28,7 @@ backends: !mux
           gce_datacenter: 'us-east1-b'
     aws: !mux
         cluster_backend: 'aws'
-        instance_type_loader: 'c4.large'
```
Comment (roydahan) [inline tests/janus-graph.yaml]: no need to change, please change back

File: tests/old/aws-longevity-1.6-1TB-7days.yaml [hunk 5]
```diff
@@ -63,4 +63,4 @@ databases: !mux
         instance_type_db: 'm3.large'
     scylla:
         db_type: scylla
-        instance_type_db: 'i2.4xlarge'
```
Comment (roydahan) [inline tests/old/aws-longevity-1.6-1TB-7days.yaml]: All tests in old directory shouldn't be changed.

File: tests/old/aws-longevity-1.6.2-1TB-7days-leveled.yaml [hunk 6]
```diff
@@ -63,4 +63,4 @@ databases: !mux
         instance_type_db: 'm3.large'
     scylla:
         db_type: scylla
-        instance_type_db: 'i2.4xlarge'
```
Comment (roydahan) [inline tests/old/aws-longevity-1.6.2-1TB-7days-leveled.yaml]: shouldn't be changed

File: tests/old/aws-sstable-1.7.yaml [hunk 7]
```diff
@@ -33,4 +33,4 @@ databases: !mux
         instance_type_db: 'm3.large'
     scylla:
         db_type: scylla
-        instance_type_db: 'i2.4xlarge'
```
Comment (roydahan) [inline tests/old/aws-sstable-1.7.yaml]: shouldn't be changed

File: tests/repair-240mins-100G.yaml [hunk 8]
```diff
@@ -60,7 +60,7 @@ backends: !mux
         cluster_backend: 'aws'
         instance_type_loader: 'c4.4xlarge'
         instance_type_monitor: 'c4.2xlarge'
-        instance_type_db: 'i2.8xlarge'
```
Comment (roydahan) [inline tests/repair-240mins-100G.yaml]: please change to i3.2xlarge

**Comment by amoskong** [comment]
I touched a New ES error: "Unknown cluster."

Test log is on aws-scylla-qa-builder1: /jenkins/amos/job-2018-10-15T08.10-f6cd74e/

```
2018-10-15 08:24:15,346 es               L0027 INFO | INDEX: growclustertest
2018-10-15 08:24:15,346 es               L0028 INFO | DOC_TYPE: test_stats
2018-10-15 08:24:15,346 es               L0029 INFO | DOC_ID: 20181015-082414-365541
2018-10-15 08:24:15,346 es               L0030 INFO | BODY: {'errors': {}, 'versions': {'scylla-jmx': {'date': '20181015', 'commit_id': '74fa
1a4', 'version': '666.development'}, 'scylla-tools': {'date': '20181015', 'commit_id': '33a5ef5739', 'version': '666.development'}, 'scylla-s
erver': {'date': '20181014', 'commit_id': '98332de26', 'version': '666.development'}}, 'setup_details': {'add_node_cnt': 1, 'test_duration': 
1440, 'cluster_backend': 'aws', 'user_prefix': 'cluster-scale-test', 'instance_type_monitor': 'm3.medium', 'cpu_platform': 'UNKNOWN', 'seeds_
num': 5, 'packages_updated': False, 'n_monitor_nodes': 1, 'subnet_id': 'subnet-ad3ce9f4', 'ami_loader_user': 'centos', 'ami_id_db_cassandra':
 'ami-3eff0028', 'cassandra_stress_threads': 500, 'ami_id_monitor': 'ami-05cfe73ca070abed2', 'ami_id_db_scylla': 'ami-05cfe73ca070abed2', 're
gion_name': 'us-east-1', 'cluster_target_size': 15, 'ami_id_loader': 'ami-05cfe73ca070abed2', 'ami_db_cassandra_user': 'ubuntu', 'n_db_nodes'
: 10, 'ami_db_scylla_user': 'centos', 'ami_monitor_user': 'centos', 'security_group_ids': 'sg-5e79983a', 'instance_type_db': 'i3.large', 'ins
tance_type_loader': 'c4.4xlarge', 'instance_provision': 'spot_low_price', 'n_loaders': 1, 'space_node_threshold': 100, 'seeds_first': 'true',
 'cassandra_stress_population_size': 10000000, 'failure_post_behavior': 'destroy', 'db_type': 'scylla', 'experimental': 'true'}, 'results': {
'throughput': defaultdict(<type 'dict'>, {})}, 'test_details': {'test_duration': 1440, 'job_url': '', 'start_time': 1539591855.334413, 'grafa
na_snapshot': '', 'sct_git_commit': 'c52a75f35f762a8d3624ffc034d0312f6e037996', 'test_name': 'grow_cluster_test.py:GrowClusterTest.test_grow_
x_to_y', 'start_host': 'ip-172-16-99-242.ec2.internal', 'job_name': 'local_run'}, 'coredumps': {}, 'nemesis': {}}
2018-10-15 08:24:15,347 retry            L0200 DEBUG| Converted retries value: False -> Retry(total=False, connect=None, read=None, redirect=
0, status=None)
2018-10-15 08:24:15,347 connectionpool   L0824 DEBUG| Starting new HTTPS connection (1): es_url_id.us-east-1.aws.found.io
2018-10-15 08:24:15,382 output           L0642 DEBUG| /usr/lib/python2.7/site-packages/urllib3/connectionpool.py:858: InsecureRequestWarning:
 Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/a
dvanced-usage.html#ssl-warnings
2018-10-15 08:24:15,383 output           L0642 DEBUG|   InsecureRequestWarning)
2018-10-15 08:24:15,400 connectionpool   L0396 DEBUG| https://es_url_id.us-east-1.aws.found.io:9243 "PUT /growclustertest HTTP/1.1" 404 41
2018-10-15 08:24:15,401 base             L0097 WARNI| PUT https://es_url_id.us-east-1.aws.found.io:9243/growclustertest [status:404 request:0
.054s]
2018-10-15 08:24:15,401 base             L0105 DEBUG| > None
2018-10-15 08:24:15,401 base             L0110 DEBUG| < {"ok":false,"message":"Unknown cluster."}
2018-10-15 08:24:15,401 tester           L0130 ERROR| Exception in setUp. Will clean resources
Traceback (most recent call last):
  File "/jenkins/slave/workspace/scylla-staging-scale-cluster-test-clone-amos/label/aws-scylla-qa-builder1/sdcm/tester.py", line 128, in wrapper
    return method(*args, **kwargs)
  File "/jenkins/slave/workspace/scylla-staging-scale-cluster-test-clone-amos/label/aws-scylla-qa-builder1/sdcm/tester.py", line 188, in setUp
    self.create_test_stats()
  File "/jenkins/slave/workspace/scylla-staging-scale-cluster-test-clone-amos/label/aws-scylla-qa-builder1/sdcm/db_stats.py", line 305, in create_test_stats
    self._create()
  File "/jenkins/slave/workspace/scylla-staging-scale-cluster-test-clone-amos/label/aws-scylla-qa-builder1/sdcm/db_stats.py", line 206, in _create
    es.ES().create(self._test_index, ES_DOC_TYPE, self._test_id, self._stats)
  File "/jenkins/slave/workspace/scylla-staging-scale-cluster-test-clone-amos/label/aws-scylla-qa-builder1/sdcm/es.py", line 31, in create
    self._create_index(index)
  File "/jenkins/slave/workspace/scylla-staging-scale-cluster-test-clone-amos/label/aws-scylla-qa-builder1/sdcm/es.py", line 20, in _create_index
    self._es.indices.create(index=index, ignore=400)
  File "/usr/lib/python2.7/site-packages/elasticsearch/client/utils.py", line 76, in _wrapped
    return func(*args, params=params, **kwargs)
  File "/usr/lib/python2.7/site-packages/elasticsearch/client/indices.py", line 91, in create
    params=params, body=body)
  File "/usr/lib/python2.7/site-packages/elasticsearch/transport.py", line 314, in perform_request
    status, headers, data = connection.perform_request(method, url, params, body, headers=headers, ignore=ignore, timeout=timeout)
  File "/usr/lib/python2.7/site-packages/elasticsearch/connection/http_urllib3.py", line 161, in perform_request
    self._raise_error(response.status, raw_data)
  File "/usr/lib/python2.7/site-packages/elasticsearch/connection/base.py", line 125, in _raise_error
    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)
NotFoundError: TransportError(404, u'{"ok":false,"message":"Unknown cluster."}')
2018-10-15 08:24:15,403 tester           L0981 DEBUG| Cleaning up resources used in the test
```

**Comment by bentsi** [comment]
Hi @amoskong,
If you think you touched a new Error please open a new issue instead of commenting on the PR that is already merged ;) 
Any way the error here is since new ES parameters were not updated in `results_analyze.yaml`

**Comment by amoskong** [comment]
Yes, that's better to track the problem by github issue, we can easy search it from there.

**Pull request merged**